index.js

```diff
@@ -2,40 +2,118 @@
 const execa = require('execa');
 const pMemoize = require('p-memoize');
 
-const device = pMemoize(() => {
-	if (process.platform !== 'darwin') {
-		return Promise.reject(new Error('macOS only'));
-	}
-
-	return execa.stdout('networksetup', ['-listallhardwareports']).then(stdout => {
-		const result = stdout.match(/Hardware Port: Wi-Fi\nDevice: (en\d)/);
+if(process.platform == 'darwin') {
 
-		if (!result) {
-			throw new Error('Couldn\'t find a Wi-Fi device');
+	const device = pMemoize(() => {
+		if (process.platform !== 'darwin') {
+			return Promise.reject(new Error('macOS only'));
 		}
 
-		return result[1];
+		return execa.stdout('networksetup', ['-listallhardwareports']).then(stdout => {
+			const result = stdout.match(/Hardware Port: Wi-Fi\nDevice: (en\d)/);
+
+			if (!result) {
+				throw new Error('Couldn\'t find a Wi-Fi device');
+			}
+
+			return result[1];
+		});
 	});
-});
 
-const isOn = device =>
-	execa.stdout('networksetup', ['-getairportpower', device]).then(stdout => /: On$/.test(stdout));
+	const isOn = device =>
+		execa.stdout('networksetup', ['-getairportpower', device]).then(stdout => /: On$/.test(stdout));
+
+	const toggleDevice = (device, turnOn) =>
+		execa('networksetup', ['-setairportpower', device, (turnOn ? 'on' : 'off')]).then(() => {});
+
+	const toggle = turnOn =>
+		device().then(device => {
+			if (typeof turnOn === 'boolean') {
+				return toggleDevice(device, turnOn);
+			}
+
+			return isOn(device).then(isOn => toggleDevice(device, !isOn));
+		});
+
+	module.exports.on = () => toggle(true);
+	module.exports.off = () => toggle(false);
+	module.exports.toggle = toggle;
+	module.exports.isOn = () => device().then(isOn);
+	module.exports.restart = () => toggle(false).then(() => toggle(true));
+	module.exports.device = device;
+} else if(process.platform == 'win32') {
+	
+	const isOn = () => {
+		let isOnline = execa.shell('ping google.com -n 1 -w 1000').then(() => {
+			return new Promise((resolve) => resolve(true));
+		}).catch(() => {
+			return new Promise((resolve) => resolve(false));
+		});
+
+		return isOnline;
+	}
+
+	const device = () => {
+		let deviceInterface = execa.shell('netsh interface show interface').then((stdout) => {
```

Comment:
Ensure you're using the `execa` API correctly. The first argument should be the binary and second should be an array of CLI arguments.